overrides/default.nix

```diff
@@ -1023,11 +1023,12 @@ lib.composeManyExtensions [
           }
         ));
 
-      gitpython = prev.gitpython.overridePythonAttrs (
-        old: {
-          buildInputs = old.buildInputs or [ ] ++ [ final.typing-extensions ];
-        }
-      );
+      gitpython = prev.gitpython.overridePythonAttrs {
+          postPatch = ''
+            substituteInPlace git/cmd.py \
+              --replace 'git_exec_name = "git"' 'git_exec_name = "${pkgs.gitMinimal}/bin/git"'
```

Comment:
gitpython should not decide on it's own to pull in a specific git as a dependency imo

Reply:
How it will function then? It is a general approach to patch packages to provide their dependencies without propagating them. You can change git version used using an overlay

Reply:
Provide the git binary in `PATH` or override the variable in your application code

Reply:
I do not expect `gitpython` the `python` package to install `git` for me

Reply:
Well, this is how gitpython in nixpkgs is packaged:

https://github.com/NixOS/nixpkgs/blob/a8be94c0af330325c8c96730cdae5e168a0632a1/pkgs/development/python-modules/gitpython/default.nix#L34

Reply:
Looks like it was introduced 6 years ago in upstream: https://github.com/NixOS/nixpkgs/pull/41844

I guess I'm fine with this behaviour then